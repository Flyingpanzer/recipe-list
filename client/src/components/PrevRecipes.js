import React, { Fragment } from 'react';

const PrevRecipes = ({ prevDesc }) => {
  return (
    <div>
      {prevDesc.length > 0 && (
        <Fragment>
          <h3>Previous versions</h3>
          <table className="table table-bordered table-striped">
            <thead>
              <tr>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              {prevDesc.map((desc, i) => {
                return (
                  <tr key={i}>
                    <td>{desc}</td>
                  </tr>
                );
              })}
            </tbody>
          </table>
        </Fragment>
      )}
    </div>
  );
};
export default PrevRecipes;
