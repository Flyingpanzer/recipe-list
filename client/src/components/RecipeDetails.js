import React, { Fragment } from 'react';
import { Button, Glyphicon } from 'react-bootstrap';

import PrevRecipes from './PrevRecipes';
const RecipeDetails = ({ recipe, showPreviousDesc, prevDesc }) => {
  return (
    <Fragment>
      {recipe && (
        <Fragment>
          <h3>Recipe Details</h3>
          <table className="table table-bordered table-striped">
            <thead>
              <tr>
                <th>
                  Description{' '}
                  <Button
                    onClick={() => showPreviousDesc(recipe)}
                    bsStyle="info"
                    bsSize="xsmall"
                  >
                    <Glyphicon glyph="th-list" />
                  </Button>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>{recipe.recipeDesc}</td>
              </tr>
            </tbody>
          </table>
        </Fragment>
      )}
      <PrevRecipes prevDesc={prevDesc} />
    </Fragment>
  );
};

export default RecipeDetails;
