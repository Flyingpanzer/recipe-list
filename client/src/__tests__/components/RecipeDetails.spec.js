import React from 'react';
import { mount } from 'enzyme';

import RecipeDetails from '../../components/RecipeDetails';
import PrevRecipes from '../../components/PrevRecipes';

const recipeDetails = {
  recipe: {
    recipeTitle: 'Omelette',
    recipeDesc: 'Desc',
  },
  showPreviousDesc: false,
  prevDesc: [' eggs, flour, and salt'],
};
describe('RecipeDetails component', () => {
  test('receive props and renders', () => {
    const wrapper = mount(<RecipeDetails recipeDetails={recipeDetails} />);

    expect(wrapper).toMatchSnapshot();
  });
});
