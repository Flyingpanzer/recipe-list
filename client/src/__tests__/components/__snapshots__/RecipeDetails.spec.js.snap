// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchDetails component receive props and renders 1`] = `
ReactWrapper {
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
    "wrapInvoke": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "recipeDetails": Object {
        "recipeFormat": "VHS",
        "recipeStars": "Alexander Kaidanovsky",
        "recipeTitle": "Stalker",
        "recipeYear": "1979	",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "recipeDetails": Object {
          "recipeFormat": "VHS",
          "recipeStars": "Alexander Kaidanovsky",
          "recipeTitle": "Stalker",
          "recipeYear": "1979	",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__linkedRoots__): Array [],
  Symbol(enzyme.__unrendered__): <RecipeDetails
    recipeDetails={
        Object {
            "recipeFormat": "VHS",
            "recipeStars": "Alexander Kaidanovsky",
            "recipeTitle": "Stalker",
            "recipeYear": "1979	",
          }
    }
/>,
  Symbol(enzyme.__updatedBy__): null,
}
`;
